src/core/server/saved_objects/migrations/core/build_active_mappings.ts

```diff
@@ -162,6 +162,14 @@ function defaultMapping(): IndexMapping {
           },
         },
       },
+      // "private" fields, not exposed through any API's
+      valid: {
+        type: 'boolean',
+      },
+      invalid_attributes: {
+        enabled: false, // Don't index invalid attributes
+        type: 'object',
+      },
```

Comment:
Should we nest all these new fields into a single nested field such as `migration` or `status` ? It may be a little cleaner?

Reply:
I've been thinking about this, it feels like something as important as the `status: 'invalid' | 'corrupt' | 'valid'` should be at the top-level rather than hidden under another layer.